pkg/config/config.go

```diff
@@ -1,155 +1,211 @@
+	// TargetNamespace is the namespace where the workloads go
+	TargetNamespace string `json:"targetNamespace,omitempty"`
 }
 
-type Hooks struct {
-	// HostToVirtual is a hook that is executed before syncing from the host to the virtual cluster
-	HostToVirtual []*Hook `json:"hostToVirtual,omitempty" yaml:"hostToVirtual,omitempty"`
-
-	// VirtualToHost is a hook that is executed before syncing from the virtual to the host cluster
-	VirtualToHost []*Hook `json:"virtualToHost,omitempty" yaml:"virtualToHost,omitempty"`
+func (v VirtualClusterConfig) Distro() string {
+	if v.Config.ControlPlane.Distro.K3S.Enabled {
+		return config.K3SDistro
+	} else if v.Config.ControlPlane.Distro.K0S.Enabled {
+		return config.K0SDistro
+	} else if v.Config.ControlPlane.Distro.K8S.Enabled {
+		return config.K8SDistro
+	} else if v.Config.ControlPlane.Distro.EKS.Enabled {
+		return config.EKSDistro
+	}
+
+	return config.K3SDistro
 }
 
-type Hook struct {
-	TypeInformation
-
-	// Verbs are the verbs that the hook should mutate
-	Verbs []string `json:"verbs,omitempty" yaml:"verbs,omitempty"`
-
-	// Patches are the patches to apply on the object to be synced
-	Patches []*Patch `json:"patches,omitempty" yaml:"patches,omitempty"`
+func (v VirtualClusterConfig) VirtualClusterKubeConfig() config.VirtualClusterKubeConfig {
+	distroConfig := config.VirtualClusterKubeConfig{}
+	switch v.Distro() {
+	case config.K3SDistro:
+		distroConfig = config.VirtualClusterKubeConfig{
+			KubeConfig:          "/data/server/cred/admin.kubeconfig",
+			ServerCAKey:         "/data/server/tls/server-ca.key",
+			ServerCACert:        "/data/server/tls/server-ca.crt",
+			ClientCACert:        "/data/server/tls/client-ca.crt",
+			RequestHeaderCACert: "/data/server/tls/request-header-ca.crt",
+		}
+	case config.K0SDistro:
+		distroConfig = config.VirtualClusterKubeConfig{
+			KubeConfig:          "/data/k0s/pki/admin.conf",
+			ServerCAKey:         "/data/k0s/pki/ca.key",
+			ServerCACert:        "/data/k0s/pki/ca.crt",
+			ClientCACert:        "/data/k0s/pki/ca.crt",
+			RequestHeaderCACert: "/data/k0s/pki/front-proxy-ca.crt",
+		}
+	case config.EKSDistro, config.K8SDistro:
+		distroConfig = config.VirtualClusterKubeConfig{
+			KubeConfig:          "/pki/admin.conf",
+			ServerCAKey:         "/pki/ca.key",
+			ServerCACert:        "/pki/ca.crt",
+			ClientCACert:        "/pki/ca.crt",
+			RequestHeaderCACert: "/pki/front-proxy-ca.crt",
+		}
+	}
+
+	retConfig := v.Config.Experimental.VirtualClusterKubeConfig
+	if retConfig.KubeConfig == "" {
+		retConfig.KubeConfig = distroConfig.KubeConfig
+	}
+	if retConfig.ClientCACert == "" {
+		retConfig.ClientCACert = distroConfig.ClientCACert
+	}
+	if retConfig.ServerCAKey == "" {
+		retConfig.ServerCAKey = distroConfig.ServerCAKey
+	}
+	if retConfig.ServerCACert == "" {
+		retConfig.ServerCACert = distroConfig.ServerCACert
+	}
+	if retConfig.RequestHeaderCACert == "" {
+		retConfig.RequestHeaderCACert = distroConfig.RequestHeaderCACert
+	}
```

Comment:
I guess its not too bad to have this explicit for now since its just 5 fields and people don't know how mergo operates and we rely on another dependency for 10 lines of code